src/script/standard.cpp

```diff
@@ -68,6 +65,21 @@ bool Solver(const CScript& scriptPubKey, txnouttype& typeRet, vector<vector<unsi
         return true;
     }
 
+    // Provably prunable, data-carrying output
+    //
+    // So long as script passes the IsUnspendable() test and all but the first
+    // byte passes the IsPushOnly() test we don't care what exactly is in the
+    // script.
+    if (scriptPubKey.size() >= 1 && scriptPubKey[0] == OP_RETURN
+                                 && scriptPubKey.size() <= nMaxDatacarrierBytes+2) // to account for the pushdata opcodes
```

Comment:
```scriptPubKey.size() <= nMaxDatacarrierBytes+2``` check belongs in policy/policy.cpp, specifically, in line https://github.com/bitcoin/bitcoin/blob/master/src/policy/policy.cpp#L52 replacing it with 
```
    } else if (whichType == TX_NULL_DATA) {
        if (scriptPubKey.size() <= nMaxDatacarrierBytes+2)
            return false;
    }
```
To do that you will need to also move nMaxDatacarrierBytes and MAX_OP_RETURN_RELAY to policy/policy.o. That would be cleaner after #6068 since the global would be moved only once directly to a CStandardPolicy attribute. But moving the condition and global already makes sense IMO.